app/web/components/CustomColorSwitch.tsx

```diff
@@ -1,52 +1,67 @@
 import { useEffect, useState } from "react";
 import { theme } from "theme";
-import makeStyles from "utils/makeStyles";
 
 import CircularProgress from "./CircularProgress";
 
-interface SwitchStyleProps {
+interface CustomSwitchProps extends Omit<SwitchProps, "color"> {
   checked: boolean;
-  color: string;
-  size: SwitchProps["size"];
+  customColor?: string; // renamed to avoid conflict with existing color prop
+  size?: SwitchProps["size"];
+  status?: string;
+  isLoading?: boolean;
 }
 
-const useStyles = makeStyles(({ palette, shadows }) => ({
-  circle: {
-    display: "flex",
-    alignItems: "center",
-    justifyContent: "center",
-    width: (props: SwitchStyleProps) => (props.size === "medium" ? 20 : 16),
-    height: (props: SwitchStyleProps) => (props.size === "medium" ? 20 : 16),
-    borderRadius: "50%",
-    backgroundColor: palette.grey[600],
-    boxShadow: shadows[1],
-  },
-  active: {
-    backgroundColor: palette.grey[600],
-  },
-  switchBase: {
-    color: palette.grey[600],
+const StyledCircle = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "size" &&
+    prop !== "checked" &&
+    prop !== "customColor" &&
+    prop !== "isLoading",
+})<CustomSwitchProps>(({ theme, size, checked, customColor, isLoading }) => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  width: size === "medium" ? 20 : 16,
+  height: size === "medium" ? 20 : 16,
+  borderRadius: "50%",
+  backgroundColor: checked ? customColor : theme.palette.grey[600],
+  boxShadow: theme.shadows[1],
+  ...(isLoading && {
+    backgroundColor: theme.palette.grey[400], // Change color when loading
+  }),
+}));
+
+const dontPassProps = ["customColor", "checked", "isLoading", "status"];
+
+const StyledSwitch = styled(Switch, {
+  shouldForwardProp: (prop) => !dontPassProps.includes(prop as string), // Filter out props that shouldn't be forwarded
+})<CustomSwitchProps>(({ theme, customColor, checked, isLoading, status }) => ({
+  "& .MuiSwitch-switchBase": {
+    color: theme.palette.grey[600],
     "& + .MuiSwitch-track": {
-      backgroundColor: palette.grey[200],
+      backgroundColor: theme.palette.grey[200],
     },
     "&.Mui-checked": {
-      color: (props: { color: string }) => props.color,
+      color: customColor,
       "& + .MuiSwitch-track": {
-        backgroundColor: (props: { color: string }) => props.color,
+        backgroundColor: customColor,
       },
     },
     "&.Mui-disabled": {
-      color: (props: { checked: boolean; color: string }) =>
-        props.checked ? props.color : palette.grey[600],
+      color: checked ? customColor : theme.palette.grey[600],
       "& + .MuiSwitch-track": {
-        backgroundColor: (props: { checked: boolean; color: string }) =>
-          props.checked ? props.color : palette.grey[200],
+        backgroundColor: checked ? customColor : theme.palette.grey[200],
         opacity: 0.4,
       },
     },
   },
+  ...(isLoading || status === "loading"
```

Comment:
Here's a preview of the new way MUI wants us to do the styles. The rest will come in a future PR.